src/server/src/db/pulse.repo.ts

```diff
@@ -22,6 +22,16 @@ export class PulseRepo {
     return res
   }
 
+  async getCategoryTimeOverview(
+    startDate: string,
+    endDate: string,
+  ): Promise<CodeClimbers.TimeOverviewDao[] | undefined> {
```

Comment:
mixing return types usually leads to issues due to having to have type checks everywhere. Would it be possible to just return an empty array instead of undefined?